src/wallet/wallet.cpp

```diff
@@ -943,6 +944,20 @@ bool CWallet::LoadUnifiedCaches()
         }
     }
 
+    // Restore decrypted Orchard notes.
+    for (const auto& [_, walletTx] : mapWallet) {
+        if (!walletTx.mapOrchardActionData.empty()) {
+            const CBlockIndex* pTxIndex;
+            std::optional<int> blockHeight;
+            if (walletTx.GetDepthInMainChain(pTxIndex) > 0) {
+                blockHeight = pTxIndex->nHeight;
+            }
+            if (!orchardWallet.RestoreDecryptedNotes(blockHeight, walletTx, walletTx.mapOrchardActionData)) {
+                return false;
```

Comment:
Add a `LogPrintf` line here like above so that we observe the cause of this failure to restore the wallet caches in the logs.

Reply:
Also, _should_ we fail here to reload the caches or just log the failure to decrypt and move on? Most of the other logic in `LoadUnifiedCaches` presents conditions where we can't recover (inconsistencies in the loaded data), but failing to decrypt isn't necessarily an example of this -- an example being maybe if an account is "removed" from the wallet (not sure if that's ever going to be possible) or there's data corruption requiring a rescan.

Reply:
I think that if we can't decrypt a note using an IVK that we previously decrypted, this is likely to have only happened due to some sort of corruption, so we can't just continue.

Reply:
Well, we expressly want to continue if there's corruption we can recover from, and I'm asking if this is an instance of that. There's a mode called salvagewallet or something that tries to read the wallet and discards the corrupted stuff.

Reply:
I don't think that wallet salvage is going to be nearly as important now once we have the ability to recover from the mnemonic seed; in the interim, I think we should be as strict as possible so that we don't miss bugs that might occur. We can't guarantee that people will look at their logs or notice if something is subtly wrong, but if their wallets stop working they'll complain loudly enough that we can fix things promptly.

Reply:
Ok but this still effectively breaks `-salvagewallet` in some cases. Please file an issue to reconsider this once we're more confident in the correctness of the Orchard wallet code.